JDI.Light/JDI.Light/Elements/Complex/Table/Table.cs

```diff
@@ -65,7 +183,164 @@ public List<UIElement> WebRow(int rowNum)
+            HeaderIsRow = firstRow.Count == 0 || Headers.Select(h => h.Text).Union(firstRow).Any();
+            return HeaderIsRow ? rowNum + 1 : rowNum;
+        }
+
+        private int GetColumnIndex(int index)
+        {
+            if (FirstColumnIndex > 1)
+            {
+                return index + FirstColumnIndex - 1;
+            }
+            return ColumnsMapping.Length > 0 ? ColumnsMapping[index - 1] : index;
+        }
+
+        public int GetColumnIndexByName(string colName)
+        {
+            var colIndex = Headers.FindIndex(h => Simplify.Invoke(h.Text).Equals(Simplify.Invoke(colName)));
+            if (colIndex == -1)
+            {
+                throw new ArgumentException($"Cannot find column {colName}");
+            }
+            return colIndex + 1;
+        }
+
+        public int GetRowIndexByName(string rowName)
+        {
+            var rowHeader = GetRowHeaderIndex() == -1 ? WebColumn(1).Select(el => el.Text).ToList() : GetRowHeader();
+            var rowIndex = rowHeader.FindIndex(h => Simplify.Invoke(h).Equals(Simplify.Invoke(rowName)));
+            if (rowIndex == -1)
+            {
+                throw new ArgumentException($"Cannot find row {rowName}");
+            }
+            return rowIndex + 1;
+        }
+
+        public List<string> GetRowHeader()
+        {
+            if (GetRowHeaderIndex() != -1) { return  WebColumn(GetRowHeaderIndex()).Select(el => el.Text).ToList(); }
+            var result = new List<string>();
+            for (var i = 1; i <= GetCount(); i++)
+            {
+                result.Add(i + "");
+            }
+            return result;
+        }
+
+        public int GetCount()
+        {
+            var firstColumn = FindElements(FillByTemplate(ColumnLocator, 1, this));
+            return firstColumn.Count;
+        }
+
+        public int GetRowHeaderIndex()
+        {
+            if (RowHeaderIndex == -1 && RowHeaderName.Equals(Empty)) { return RowHeaderIndex; }
+            var index =  Headers.FindIndex(h => Simplify.Invoke(h.Text).Equals(Simplify.Invoke(RowHeaderName)));
+            if (index > -1)
+            {
+                RowHeaderIndex = index + 1;
+            }
+            else
+            {
+                throw new ArgumentException($"Cannot find rowHeader {RowHeaderName} in 'header' {Join("", Headers.Select(h => h.Text))}. Please correct params");
+            }
+            return RowHeaderIndex;
+        }
+
+        public static By FillByTemplate(By by, params object[] args)
+        {
+            var byLocator = GetByLocator(by);
+            try
+            {
+                byLocator = Format(byLocator, args);
+            }
+            catch (FormatException)
+            {
+                throw new FormatException("Bad locator template '" + byLocator + "'. Args: " + Join("", args));
+            }
+            return By.XPath(byLocator);
+        }
+
+        public static string GetByLocator(By by)
```

Comment:
Please do not create a static members inside a class that can have several instances.
Different tables should be able to have different implementations of this method.